upup/pkg/fi/cloudup/alitasks/sshkey.go

```diff
@@ -68,13 +70,40 @@ func (s *SSHKey) Find(c *fi.Context) (*SSHKey, error) {
 		Name:               fi.String(k.KeyPairName),
 		KeyPairFingerPrint: fi.String(k.KeyPairFingerPrint),
 	}
+
+	// Avoid spurious changes
+	if fi.StringValue(actual.KeyPairFingerPrint) == compactKeyFingerprint(fi.StringValue(s.KeyPairFingerPrint)) {
```

Comment:
Usually a KeyFingerprint looks like: "a7:b1:3e:3d:84:24:a2:5a:91:5f:6f:e9:cf:dd:2b:6a".
For unknown reason, when creating new ssh key, it returns key print without ":", eg: "a7b13e3d8424a25a915f6fe9cfdd2b6a".

But in describe sshkey method, it returns normal KeyFingerprint. So I made this `compactKeyFingerprint` function to do comparison.